pkg/profilecreator/profilecreator.go

```diff
@@ -266,31 +295,38 @@ func (ghwHandler GHWHandler) GetReservedAndIsolatedCPUs(reservedCPUCount int, sp
 		} else {
 			max = reservedCPUCount
 		}
-		//sorting by comparing the first logical core of processor cores
-		sort.Slice(node.Cores, func(i, j int) bool {
-			//sorting the logical cores a processor
-			sort.Slice(node.Cores[i].LogicalProcessors, func(x, y int) bool {
-				return node.Cores[i].LogicalProcessors[x] < node.Cores[i].LogicalProcessors[y]
-			})
-			sort.Slice(node.Cores[j].LogicalProcessors, func(x, y int) bool {
-				return node.Cores[j].LogicalProcessors[x] < node.Cores[j].LogicalProcessors[y]
-			})
-			return node.Cores[i].LogicalProcessors[0] < node.Cores[j].LogicalProcessors[0]
-		})
+		if max%2 != 0 && htEnabled {
+			return "", "", fmt.Errorf("Can't allocatable odd number of CPUs from a NUMA Node")
+		}
 		for _, cores := range node.Cores {
 			for _, core := range cores.LogicalProcessors {
```

Comment:
I see 2 variables with the same name, "cores", while "it works" maybe it worth renaming one of them

Reply:
Sure, will fix.